src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -73,78 +73,81 @@ public void SerializeDebugInfo(IMethodBody methodBody, StandaloneSignatureHandle
             // Special case a hidden entry point (#line hidden applied) that would otherwise have no debug info.
             // This is to accommodate for a requirement of Windows PDB writer that the entry point method must have some debug information.
             bool isKickoffMethod = methodBody.StateMachineTypeName != null;
-            bool emitDebugInfo = isKickoffMethod || !methodBody.SequencePoints.IsEmpty ||
+            bool emitAllDebugInfo = isKickoffMethod || !methodBody.SequencePoints.IsEmpty ||
                 methodBody.MethodDefinition == (Context.Module.DebugEntryPoint ?? Context.Module.PEEntryPoint);
 
-            if (!emitDebugInfo)
+            var compilationOptions = Context.Module.CommonCompilation.Options;
+
+            // We need to avoid emitting CDI DynamicLocals = 5 and EditAndContinueLocalSlotMap = 6 for files processed by WinMDExp until 
+            // bug #1067635 is fixed and available in SDK.
+            bool suppressNewCustomDebugInfo = compilationOptions.OutputKind == OutputKind.WindowsRuntimeMetadata;
+
+            bool emitDynamicAndTupleInfo = emitAllDebugInfo && !suppressNewCustomDebugInfo;
+
+            // Emit EnC info for all methods even if they do not have sequence points.
+            // The information facilitates reusing lambdas and closures. The reusal is important for runtimes that can't add new ones (e.g. Mono).
+            bool emitEncInfo = compilationOptions.EnableEditAndContinue && _metadataWriter.IsFullMetadata && !suppressNewCustomDebugInfo;
+
+            byte[] blob = customDebugInfoWriter.SerializeMethodDebugInfo(Context, methodBody, methodHandle, emitStateMachineInfo: emitAllDebugInfo, emitEncInfo, emitDynamicAndTupleInfo, out bool emitExternNamespaces);
```

Comment:
If I'm reading this right, this changes so the custom debug info is never written for WindowsRuntimeMetadata? Seems like the previous code would write some of it (state machines for example)